File: 02/main.py
with open("input.txt") as f:
    data = []
    for x in f.read().strip().split("\n"):
        x = x.split(" ")
        data.append((x[0], int(x[1])))

position = 0
depth = 0
aim = 0

for c in data:
    if c[0] == "forward":
        position += c[1]
        depth += c[1] * aim
    elif c[0] == "down":
        aim += c[1]
    elif c[0] == "up":
        aim -= c[1]
    else:
        assert False, "unreachable"

print(position * depth)

File: 02/main.cpp
#include <fstream>
#include <iostream>

using namespace std;

int main() {
    ifstream fp("input.txt");

    int pos = 0, depth = 0;

    while (true) {
        string direction;
        int amount;

        if (!(fp >> direction)) {
            break;
        }
        fp >> amount;

        if (direction == "forward") {
            pos += amount;
        }else if (direction == "up") {
            depth -= amount;
        }else if (direction == "down") {
            depth += amount;
        }
    }

    cout << pos * depth << "\n";
    return 0;
}